rules/preprocessing.smk

```diff
@@ -0,0 +1,65 @@
+import os
+import glob
+# Documentation:
+# Workflow that involves preprocessing raw single-cell plate data and
+# transforming it into normalized aggregate profiles.
+#
+# Parameters
+# ---------
+# sql_files : List[str]
+#   List of SQL files containing plate data
+# barcodes : str
+#   path pointing to the
+# metadata : str
+#   path pointing to plate metadata
+#
+# Generates
+# -------
+# cell_counts: .csv file
+#   csv file containing n_cells per well
+# augmented: csv.gz file
+#   Annotated aggregated profiles
+# normalized: csv.gz file
+#   Normalized annotated aggregate profiles
+#--------------------
+
+
+# collecting all unqieuids from plate
+configfile: "configs/configuration.yaml"
+rule aggregate:
+    input:
+        sql_files=expand("data/{plate_id}.sqlite", plate_id=PLATE_IDS),
+        barcodes="data/barcode_platemap.csv",
+        metadata="data/metadata"
+    output:
+        cell_counts=expand("results/preprocessing/{plate_id}.cell_counts.tsv", plate_id=PLATE_IDS),
+        aggregate_profile=expand("results/preprocessing/{plate_id}.aggregate.csv.gz", plate_id=PLATE_IDS)
```

Comment:
```suggestion
        aggregate_profile=expand("results/preprocessing/{plate_id}_aggregated.csv.gz", plate_id=PLATE_IDS)
```

Reply:
I don't see aggregated.csv.gz defined in the snake file,  while annotated and normalized are. Does it matter?

Reply:
are you referring to the `.smk`  file of the actual `Snakefile` .


The way how it snakemake works is that in the `Snakefile` you specify the expected outputs that our workflow will produce. Therefore, all paths written under the `input` rule attribute in the `Snakefile` should match with the `ouput` rule attribute in our `.smk` files.

So to answer you question it does matter. From the looks of it both `aggregated.csv.gz` are specified in the `Snakefile` and `.smk` file

Reply:
thanks!